### ISSUE TITLE
Is there any plan to support filter log or similar API?

### ISSUE THREAD

**Issue by Haoxiqiang** [brave/adblock-rust#397]
When using multiple rule sets, it is difficult to know which rules are currently in effect when adjusting them. Consider supporting a filter log similar to `AdGuard` or `uBlock Origin`, as shown below:
![Image](https://github.com/user-attachments/assets/5bd92182-efd2-4714-ac1d-bca93b2b7449)

![Image](https://github.com/user-attachments/assets/6b750cee-041d-4740-949d-192144a5919f)

I found `adblock-rust` contains the information.
```
pub struct NetworkFilter {
    ...
    pub hostname: Option<String>,
    pub(crate) tag: Option<String>,
    pub raw_line: Option<Box<String>>,
    ...
}
```
Can we provide a mechanism to extend BlockResult or have a completely new API implementation to support it?

**Comment by antonok-edm** [comment]
Yes, it is planned, but will require some significant changes. The way we have right now in `BlockResult` is not ideal; storing a copy of the rule string wastes a lot of runtime memory. That's why we currently don't enable it by default.

This should ultimately be implemented by storing a line number so that the original rule can be retrieved from the filter list as needed (i.e. only when a logger is opened).
